src/System.Text.Json/tests/Serialization/TestClasses.SimpleTestClassWithObjectArrays.cs

```diff
@@ -24,6 +24,7 @@ public class SimpleTestClassWithObjectArrays : ITestClass
         public object[] MySingle { get; set; }
         public object[] MyDouble { get; set; }
         public object[] MyDateTime { get; set; }
+        public object[] MyUri { get; set; }
```

Comment:
Please add some negative tests like `Parse<Uri>(@"""{}""")` and to make sure JsonException is being thrown from the converter.